foundation/foundation-upgrade/src/test/java/com/boozallen/aissemble/upgrade/migration/version_specific/SparkBomDependencyMigrationStep.java

```diff
@@ -0,0 +1,83 @@
+package com.boozallen.aissemble.upgrade.migration.version_specific;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractMigrationTest;
+import com.boozallen.aissemble.upgrade.migration.extensions.SparkBomDependencyMigrationTest;
+
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.When;
+import io.cucumber.java.en.But;
+import io.cucumber.java.en.Then;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Parent;
+import org.technologybrewery.baton.util.pom.PomHelper;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+public class SparkBomDependencyMigrationStep extends AbstractMigrationTest {
+
+    @Given("a POM file that doesn't have the aissemble-spark-bom dependency")
+    public void aPomFileWithoutAissembleSparkBom() {
+       setTestFileToVersionMigration("SparkBomDependencyMigration", "pom.xml");
+    }
+
+    @Given("a POM file that doesn't have the aissemble-spark-bom dependency with no spark dependency")
+    public void aPomFileWithoutAissembleSparkBomAndNoSparkDependency() {
+       setTestFileToVersionMigration("SparkBomDependencyMigration", "pom-no-spark.xml");
+    }
+
+    @Given("a POM file that doesn't have the aissemble-spark-bom dependency with an existing dependencyManagement section")
+    public void aPomFileWithoutAissembleSparkBomAndExistingDependencyManagementSection() {
+       setTestFileToVersionMigration("SparkBomDependencyMigration", "pom-dependency-management-exists.xml");
+    }
+
+    @And("contains a spark dependency")
```

Comment:
a: Like to keep the steps file as given, when, then for clarity
```suggestion
    @Given("contains a spark dependency")
```

Reply:
q: If this is a given then why are we asserting?

Reply:
Simplified steps file to remove assertions for "And" steps since the logic for that is handled in the migration itself